upgrade-agent/install.go

```diff
@@ -0,0 +1,299 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+package upgradeagent
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"context"
+	"crypto/sha256"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"net/http"
+	"os"
+	"path"
+	"time"
+
+	"github.com/edgelesssys/constellation/v2/internal/retry"
+	"github.com/edgelesssys/constellation/v2/internal/versions"
+	"github.com/spf13/afero"
+	"k8s.io/utils/clock"
+)
+
+const (
+	// determines the period after which retryDownloadToTempDir will retry a download.
+	downloadInterval = 10 * time.Millisecond
+)
+
+// osInstaller installs binary components of supported kubernetes versions.
+type osInstaller struct {
+	fs      *afero.Afero
+	hClient httpClient
+	// clock is needed for testing purposes
+	clock clock.WithTicker
+	// retriable is the function used to check if an error is retriable. Needed for testing.
+	retriable func(error) bool
+}
+
+// newOSInstaller creates a new osInstaller.
+func newOSInstaller() *osInstaller {
```

Comment:
This is more code than I remembered we had for this. I think when we remove then `.tar` part (see comment below), this should be a lot less. Still, I'm on the fence whether this is still considered ["a little copying"](https://go-proverbs.github.io/). What are your opinions @katexochen @stdoutput: should we move the installer from `bootstrapper/internal` to `internal` and share the code?

Reply:
Yes, I agree.

Reply:
IMO, yes (move it)